service/history/task/processor.go

```diff
@@ -172,3 +246,28 @@ func createTaskScheduler(
 
 	return scheduler, err
 }
+
+func getDomainPriorityWeight(
+	logger log.Logger,
+	config *config.Config,
+	domainCache cache.DomainCache,
+	k DomainPriorityKey,
+) int {
+	var weights map[int]int
+	domainName, err := domainCache.GetDomainName(k.DomainID)
+	if err != nil {
+		logger.Error("failed to get domain name from cache, use default round robin weights", tag.Error(err))
+		weights = dynamicconfig.DefaultTaskSchedulerRoundRobinWeights
+	} else {
+		weights, err = common.ConvertDynamicConfigMapPropertyToIntMap(config.TaskSchedulerDomainRoundRobinWeights(domainName))
+		if err != nil {
+			logger.Error("failed to convert dynamic config map to int map, use default round robin weights", tag.Error(err))
+			weights = dynamicconfig.DefaultTaskSchedulerRoundRobinWeights
+		}
+	}
+	weight, ok := weights[k.Priority]
+	if !ok {
+		logger.Error("weights not found for task priority", tag.Dynamic("priority", k.Priority), tag.Dynamic("weights", weights))
+	}
+	return weight
```

Comment:
is defaulting to 0 the right thing when `weights` doesn't have the given priority key?